### ISSUE TITLE
RMSD reference containing protein and ligand

### ISSUE THREAD

**Issue by bcossins** [OpenBioSim/biosimspace#376]
**Is your feature request related to a problem? Please describe.**
I want to use BioSimSpace.Metadynamics.CollectiveVariable.RMSD on a ligand pose

**Describe the solution you'd like**
I would like to be able to use BioSimSpace.Metadynamics.CollectiveVariable.RMSD with a ligand pose in its pocket which means a reference containing more than one molecule

**Describe alternatives you've considered**
I don't see an alternative to using a multi-molecule reference when using RMSD CVs with ligand poses

**Additional context**

**Comment by lohedges** [comment]
This should be easy to do. The collective variable was actually added for steered molecular dynamics, not metadynamics. If I recall we need to provide a reference PDB at some point, so there is the need to map indices correctly. I think the logic would still carry over to a multi molecule system.

**Comment by lohedges** [comment]
For now you can just create two CVs and use them both.

**Comment by lohedges** [comment]
Actually, that won't work since the same name is used for the reference PDB. This is a bug, so I'll fix tomorrow.

**Comment by bcossins** [comment]
Yea have been trying it but at the moment the interface blocks me from using another other than a single molecule as a reference.

I want to use a protein-ligand structure as a reference with two molecules.
Needs to be one RMSD CV that aligns on protein and ligand and then gives RMSD on ligand only.

**Comment by lohedges** [comment]
I think you'll need to use two CVs then, otherwise there is no way to decompose, i.e. for the purposes of measuring one the ligand RMSD only.

**Comment by bcossins** [comment]
It wouldnt work with two CVs.
Both openMM and plumed can calculate RMSD on a subset of the alignment atom group.

**Comment by lohedges** [comment]
Okay, then we'll need a different API then. Will have a think.

**Comment by bcossins** [comment]
Sure, no worries or rush.
For now I am doing everything in BSS but then converting to openMM and filling the context with the bits needed for protein-ligand metaD that way around. Hope this will work.

This sort of thing https://sire.openbiosim.org/cheatsheet/openmm.html#

**Comment by lohedges** [comment]
One thing I'll add is the ability to specify a mapping between the  molecule(s) used in the reference and the system so that the user doesn't need to pass a full system as the reference. At the moment it assumes that the two systems match, i.e. have the same number of molecules in the same order, etc.

**Comment by bcossins** [comment]
Sounds good.
Can BSS.Metadynamics.CollectiveVariable.RMSD be used with openMM using RMSDforce and BiasVariable?

**Comment by lohedges** [comment]
No, the only CV that currently works with OpenMM is Funnel. This is because we aren't using Plumed so the OpenMM scripts are completely custom. All CVs should work with Plumed patched engines, e.g. AMBER or GROMACS.
